Bas stu2 prepublish review

isaacvetter opened the pull request:
1) Lot's of small/non-functional wording changes / improvements per Bas and #351 
2) Rename "Draft - STU2" to "Current - STU2" and clean up menu
3) point to jira as location for submitting issues per #352

wmaethner commented on docs/specification/STU2.md:
```diff
@@ -125,30 +128,30 @@ Field | Optionality | Channel | Type | Description
 ### `webhook` vs `websocket`
 
-A subscriber specifies the preferred `hub.channel.type` of either `webhook` or `websocket` during creation of its subscription. Subscribers SHOULD use websockets when they are unable to host an accessible callback URL.
+A Hub SHALL support WebSockets and MAY support webhooks subscriptions. A subscriber specifies the preferred `hub.channel.type` of either `webhook` or `websocket` during creation of its subscription. Subscribers should use websockets when they are unable to host an accessible callback URL.
```
Is there a reason for not capitalizing this "should"? It is capitalized in other places making it seem like one of HL7's special verbs. If it isn't meant to be an "official" **should** in this case, can we use a different verb so as not to be confusing

isaacvetter replied:
Thanks, Will! Changed sentence to:
>Websockets are particularly useful if a subscriber is unable to host an accessible callback URL.

wmaethner reviewed (changes requested):
Small question/comment

isaacvetter commented:
Happy New Year, @bvdh !

Okay, of these four issues:

1. >Presence of '-' in proprietary events - I suggest to file an issue on this and discuss it as part of STU3

Yes, will you please log this as a jira issue? (If one doesn't already exist?)

2. >Subscription request format - last box related to websockets - "Required when hub.channel.type=websocket for re-subscribes and unsubscribes." is a duplication of the column that states it is conditional for webscockets. The sentence on its own is not wrong - remove it or make it an issue for STU3.

Will you please log this as a jira issue? I'm likely misunderstanding, because I'm uncertain you're right. Note that not only is `hub.channel.type` required for websockets and not for webhooks, but more specifically is not required for initial subscribes when using websockets. Elsewhere, when we use CONDITIONAL, it typically means that websockets always requires and webhooks doesn't or vice-versa.

1. >The computable syntax of FHIRcast events in section (https://github.com/HL7/fhircast-docs/blob/8200aa1a22201ab553f81a15bce6ddce02e3de86/docs/specification/STU2.md#fhircast-authorization--smart-scopes) does not correspond to the event definition. I suggest that we remove it, it has no impact on the spec and will prevent confusion.

In [this section](http://fhircast.org/specification/STU2/#fhircast-authorization-smart-scopes), there are two diagrams. The first is introduced with this phrase: "the FHIRcast syntax for workflow related events is:", and the second with: "The FHIRcast event name is also a computable syntax, the complete syntax for FHIRcast scopes is:". Yet, the first diagram describes only OAuth scope syntax and the second diagram describes the event name format within an OAuth2 scope. The text introducing the first image is wrong. Changed to: "the FHIRcast syntax for OAuth 2.0 access scopes is:".

Secondly, since we've also added an additional diagram describing EBNF for actual events (in the [Event Definition Format hook name section](http://fhircast.org/specification/STU2/#event-definition-format-hook-name)). I'm also removing the full EBNF which attempts to document both the event name and OAuth2 scope EBNF structure.

1. >There are still an issue I really do not agree with. It is related to events being stateless (https://github.com/HL7/fhircast-docs/blob/8200aa1a22201ab553f81a15bce6ddce02e3de86/docs/specification/STU2.md#events). This is not true. I cannot derive the current state of a session from a close event. I would like it to be so but the current spec works differently. The only way to make this happen is when only close events are allowed for currently open anchor resources. But this would go further than an editorial change. Therefore I suggest that we remove the sentence from the spec and create an issue on this to be resolved in STU3.

Bas, I think we're debating the meaning of words and not the intent or understanding. I updated the sentence: "FHIRcast events are stateless" to "FHIRcast events do not communicate previous state". Do you agree with this change? If not, let's remove the sentence. 

Isaac

bvdh commented:
Allow for '-' to be present in proprietary FHIRcast events 
==> https://jira.hl7.org/browse/FHIR-30210?filter=-2

Subscription request format 
==> https://jira.hl7.org/browse/FHIR-30211

I updated the sentence: "FHIRcast events are stateless" to "FHIRcast events do not communicate previous state". Do you agree with this change? If not, let's remove the sentence.
-- I Agreee
